### DISCUSSION TITLE
Next Router pathname is incorrect

### DISCUSSION THREAD

**Discussion by j-quelly** [vercel/next.js#13526] [Help]
# Bug report

## Describe the bug

After implementing a custom server with express.js and trying to navigate to the path `/docs-beta` the what-I-suspect-is-the-client-side-router redirects to `/` (root).  I've tried adding a route to the server.js, but to no avail:
```
    server.get('/docs-beta', (req, res) => {
      return app.render(req, res, '/docs-beta', req.query);
    });
```
I should note that `/docs-beta` uses an `index.mdx` file  and `pageExtensions: ['js', 'jsx', 'md', 'mdx', 'tsx', 'ts'],` is set in `next.config.js`

## To Reproduce

**Server.**
I am starting the server via `nodemon --inspect server/index.js`:
```
import express from 'express';
import next from 'next';
import { PORT, IS_DEV, IS_PROD } from './config';
import routes from './routes';

const app = next({ dev: IS_DEV });
const handle = app.getRequestHandler();

app.prepare().then(() => {
  const server = express();

  // Default catch-all handler to allow Next.js to handle
  // all other routes not specifically defined
  server.all('*', (req, res) => handle(req, res));

  server.listen(PORT, err => {
    if (err) throw err;
    console.log(`> Ready on http://localhost:${PORT}`);
  });
});
```
**nodemon.json**
```
{
    "verbose": true,
    "ignore": [
        "src",
        "public"
    ]
}
```
**_app.tsx**
```
import dynamic from 'next/dynamic';
import React from 'react';
import { withRouter } from 'next/router';

import '../scss/vendor/_reset.scss';

import DocsBeta from '../components/docs-beta/DocsBeta';

const App = props => {
  const { Component, pageProps, router } = props;

  const DOCS_BETA_PATH = /\/docs-beta(.*)/;
  if (!DOCS_BETA_PATH.test(router.pathname)) {
    return <Component {...pageProps} />;
  } else {
    return <DocsBeta {...props} />;
  }
};

// monkeypatch for webpack HMR issue
// https://github.com/zeit/next-plugins/issues/282#issuecomment-594088880
// https://github.com/sheerun/extracted-loader/issues/11
if (module.hot) {
  module.hot.addStatusHandler(status => {
    if (typeof window !== 'undefined' && status === 'ready') {
      // @ts-ignore
      window.__webpack_reload_css__ = true;
    }
  });
}

App.displayName = 'App';
export default withRouter(App);
```

When trying to navigate to `http://localhost:8003/docs-beta` you'll see that `router.pathname` value is `/` in `_app.tsx` instead of `/docs-beta`. 

I've manually test navigating to other routes such as `/products` and `router.pathname` is correct in `_app.tsx` whereas when `/docs-beta` is entered into the address bar and I press enter, the value value is `/` in `_app.tsx`. I'm also able to successfully load `/docs-beta/some-other-path` without any problems.

**Update**
I've tried to reproduce in Safari and everything works as expected.

## Expected behavior

`router.pathname` should be `/docs-beta`

## Screenshots

![Screen Shot 2020-05-28 at 5 53 20 PM](https://user-images.githubusercontent.com/10458851/83209944-65567380-a10e-11ea-94e3-ae9f599563a9.png)

## System information

- OS: macOS
- Browser (if applies) Chrome Version 83.0.4103.61 (Official Build) (64-bit)
- Version of Next.js: 9.2.2
- Version of Node.js: v12.16.2

**Comment by timneutkens** [comment]
You're probably looking to use `router.asPath` which is the displayed path (not the page path)

**Reply by j-quelly** [reply]
> You're probably looking to use `router.asPath` which is the displayed path (not the page path)

No, sorry, still the same results:

![Screen Shot 2020-05-29 at 8 27 25 AM](https://user-images.githubusercontent.com/10458851/83277076-65925580-a186-11ea-884f-a6df66c4efb5.png)

Could it be that there is no `/pages/docs-beta.mdx` file? Instead we have `/pages/docs-beta/index.mdx` would that matter?

**Comment by abeninski** [comment]
I have the exact same issue. Is there a fix? Router is getting out of sync
